### PR TITLE
LPS-182691 Set the AMImageMagickImageScaler as default

### PR DESCRIPTION
**Pull request by robertoDiaz**
We have a ton of cases related with AM supported image types that were solved the same way: ImageMagickImageScaler

With current implementation we have to be reactive and add a new mime type to the service component with every customer new issue.

I've change the way it is registered and add new logic to allow customers add new image types if needed with a property.

Thanks @SamZiemer for helping me testing this fix asserting the all the current mime types work as expected

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/scaler/AMImageMagickImageScaler.java [hunk 1]
```diff
@@ -43,12 +47,10 @@
...
 /**
  * @author Eric Yan
+ * @author Roberto Díaz
  */
 @Component(
-	property = {
-		"mime.type=image/gif", "mime.type=image/heic", "mime.type=image/tiff",
-		"mime.type=image/webp"
-	},
+	property = {"mime.type=*", "service.ranking:Integer=101"},
```
Comment (adolfopa) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/scaler/AMImageMagickImageScaler.java:53]: The default service ranking is 0; for consistency it is better to use the value 100 as in other places.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are unrelated

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:forward

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**